source/common/protobuf/utility.cc

```diff
@@ -656,8 +656,19 @@ void MessageUtil::unpackTo(const ProtobufWkt::Any& any_message, Protobuf::Messag
                                          any_message_with_fixup.DebugString()));
       }
       Config::VersionConverter::annotateWithOriginalType(*earlier_version_desc, message);
-      MessageUtil::onVersionUpgradeDeprecation(any_full_name);
-      return;
+      // To exclude rejecting below v2 protos.
+      std::set<absl::string_view> exclude_v2_protos{
+          "envoy.config.health_checker.redis.v2",
```

Comment:
Thanks for iterating, I think this approach is the best to work with for now.

I recommend putting this exclusion list in `api_type_oracle.cc` and applying to all 3 variants. You want to use the v3 types there. Also, please write some unit/integration tests to demonstrate that for at least one of these that you can successfully use v2 and v3 without triggering the `runtime.deprecated_feature_seen_since_process_start` stat.
/wait

Reply:
Ok, Sure.  let me do it now.

Reply:
changes done for `api_type_oracle.cc` . understanding to write unit tst for it.